packages/common/src/PreventExit/store.ts

```diff
@@ -0,0 +1,55 @@
+const keys: Array<symbol> = [];
+const allPreventExitState: Record<symbol, boolean> = {};
+
+function checkNoMorePreventExit() {
+  if (keys.map((key) => allPreventExitState[key]).every((i) => i === false)) {
+    window.onbeforeunload = null;
+    return true;
+  }
+  return false;
+}
+
+function publish(id: symbol, preventExit: boolean) {
+  allPreventExitState[id] = preventExit;
+  if (!checkNoMorePreventExit())
+    window.onbeforeunload = (e) => {
+      e.preventDefault();
+      return 'Confirm Alert appears';
+    };
+}
+
+interface PreventExitListenerResult {
+  /**
```

Comment:
can you add TSDoc and export this interface.

Reply:
@arunachalam-monk you did not fix this!!!!!!!!!!!!!